Lib/test/test_importlib/extension/test_loader.py

```diff
@@ -145,6 +146,15 @@ def test_reload(self):
             importlib.reload(module)
             self.assertIs(ex_class, module.Example)
 
+    def test_reload_faliure(self):
+        '''Test that reload throws ModuleNotFounderror if reloading'''
+        '''a module whose spec cannot be found'''
+        datetime.__name__ = 'destroyed_datetime'
```

Comment:
We probably shouldn't directly mess with a stdlib module and `sys.modules` and use a helper like Nick said. Untested, but I think we need to add something like the following snippet to `Lib/test/test_importlib/test_api.py`:

```py
with support.CleanImport('types'):
    module = self.init.import_module('types')
    module.__name__ = 'broken_types'
    module.__spec__ = None
    sys.modules[module.__name__] = module  # TODO: Do we need to cleanup sys.modules?
    with self.assertRaises(ModuleNotFoundError):
        self.init.reload(module)
```